Sources/Common/Core/DataArray/index.js

```diff
@@ -299,8 +298,10 @@ function vtkDataArray(publicAPI, model) {
     return publicAPI.insertTuples(idx, tuples);
   };
 
-  publicAPI.getTuple = (idx, tupleToFill = TUPLE_HOLDER) => {
+  publicAPI.getTuple = (idx, tupleToFill) => {
```

Comment:
How about replacing this with `tupleToFill = []`? It will create a new array instance every invocation, unlike the previous implementation.

Reply:
which could kill the perf when done in a loop.

Reply:
This is indeed a "perf" vs "hard-to-debug" dilemna. Here are some real-life code examples that would be buggish:
```
foo(da.getTuple(0), da.getTuple(1), da.getTuple(2));
vtkMath.areEquals(da1.getTuple(0), da2.getTuple(0));
```

To me, if performance is an issue, `tupleToFill` MUST be provided by caller (to be documented in index.d.ts).

Reply:
We could add docs strongly recommending `tupleToFill` be passed in by the user for perf reasons, but leave the signature as `tupleToFill = []` to have a sane default. Additionally, we should rewrite all internal invocations of `getTuple()` to pass in an array.

Reply:
Let's go with no bug... And perf can be fixed if arg is passed... I guess it is better that way